### PR TITLE
Fix British English Date Format

### PR DESCRIPTION
**Pull request by rom-1**
The default Date Format for British English is actually not like the american one (M/D/Y), but more like the same as the rest of Europe (D/M/Y). As FreshRSS offers a translation file explicitly for American English (en-US), this one should be considered as Oxford/British english.

### CODE REVIEW

**Comment by rom-1** [comment]
whoops, some checks failed? Sorry, did not want to break anything...

**Comment by Alkarex** [comment]
@rom-1 Don't worry about the Docker tests, the important one for this type of PR is Travis

**Comment by Alkarex** [comment]
Is the dot after the day intentional?

image

Cf. https://englishlessonsbrighton.co.uk/how-to-write-dates-british-american-english/

**Comment by Frenzie** [comment]
@Alkarex Yeah, but I fixed the Travis test. ;-)

**Comment by Alkarex** [comment]
By the way, for another PR if someone could take it, I have just noticed that the layout is slightly broken with most themes for longer en-US dates:
image

**Comment by rom-1** [comment]
> Is the dot after the day intentional?
> 
> image
> 
> Cf. https://englishlessonsbrighton.co.uk/how-to-write-dates-british-american-english/

It is not, good find!

**Comment by rom-1** [comment]
> By the way, for another PR if someone could take it, I have just noticed that the layout is slightly broken with most themes for longer en-US dates:
> image

the en-US dates are broken anyway, as US does not have 24h time format -> 12h am/pm

File: app/i18n/es/gen.php [hunk 1]
```diff
@@ -72,8 +72,8 @@
 		'december' => 'diciembre',
 		'feb' => 'feb',
 		'february' => 'febrero',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',	// TODO - Translation
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',	// TODO - Translation
+		'format_date' => 'j\\. %s Y',
+		'format_date_hour' => 'j\\. %s Y \\a\\t H\\:i',
```
Comment (Frenzie) [inline app/i18n/es/gen.php]:
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

File: app/i18n/en/gen.php [hunk 2]
```diff
@@ -72,8 +72,8 @@
 		'december' => 'December',
 		'feb' => 'Feb.',
 		'february' => 'February',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',
+		'format_date' => 'j\\ %s Y',
+		'format_date_hour' => 'j\\ %s Y \\a\\t H\\:i',
```
Comment (Frenzie) [inline app/i18n/en/gen.php]: That \\ isn't for anything anymore, is it?
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

File: app/i18n/he/gen.php [hunk 3]
```diff
@@ -72,8 +72,8 @@
 		'december' => 'Dec',
 		'feb' => 'feb',
 		'february' => 'Feb',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',	// TODO - Translation
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',	// TODO - Translation
+		'format_date' => 'j\\ %s Y',
+		'format_date_hour' => 'j\\ %s Y \\a\\t H\\:i',
```
Comment (Frenzie) [inline app/i18n/he/gen.php]:
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

File: app/i18n/ru/gen.php [hunk 4]
```diff
@@ -72,8 +72,8 @@
 		'december' => 'Dec',
 		'feb' => 'feb',
 		'february' => 'Feb',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',	// TODO - Translation
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',	// TODO - Translation
+		'format_date' => 'j\\. %s Y',
+		'format_date_hour' => 'j\\. %s Y \\a\\t H\\:i',
```
Comment (Frenzie) [inline app/i18n/ru/gen.php]:
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

File: app/i18n/tr/gen.php [hunk 5]
```diff
@@ -72,8 +72,8 @@
 		'december' => 'Ara',
 		'feb' => 'şub',
 		'february' => 'Şub',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',	// TODO - Translation
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',	// TODO - Translation
+		'format_date' => 'j\\. %s Y',
+		'format_date_hour' => 'j\\. %s Y \\a\\t H\\:i',
```
Comment (Frenzie) [inline app/i18n/tr/gen.php]:
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

File: app/i18n/en/gen.php (hunk 2 above)
Comment (rom-1) [reply app/i18n/en/gen.php]: hope so... :D

**Comment by rom-1** [comment]
ok, so I added more commits, removing the dot, and also fixed the american time (24h -> 12h). As I have almost no experience using GIT, just my simple question: Is there anything else I need to do right now to commit all these changes? Or is everything alright from my side, and you guys just have to approve it?

**Comment by Alkarex** [comment]
@rom-1 You did it just right, thanks. Next steps are review + test + merge from our side.

**Review by Alkarex — APPROVED** [review]

**Review by Alkarex — CHANGES_REQUESTED** [review]
There is a little warning on the Travis automatic test:
https://travis-ci.org/github/FreshRSS/FreshRSS/jobs/752642159

**Comment by Alkarex** [comment]
Because `format_date` is now identical in `en` than some other languages, the corresponding "ignore" files must be updated

**Comment by rom-1** [comment]
> Because `format_date` is now identical in `en` than some other languages, the corresponding "ignore" files must be updated

I could simply copy Frenzie's work and add 2 lines ('gen.date.format_date', 'gen.date.format_date_hour',) to the array of every cli/i18n/ignore/xx.php file. But while this might be the correct way to fix it, I would have no idea what I am doing. So maybe I am not the best person changing this?

**Comment by aledeg** [comment]
You can do it with the help of the make command

**Comment by rom-1** [comment]
> You can do it with the help of the make command

you are by far overestimating my skills! :)
I have never used make in the past nor have a linux/unix machine at hand. 

I manually added Frenzie's 2 lines to the array of every ignore-language-file except of the en.php one. As stated, I have no idea what I am doing here, but maybe this will fix the travis checks.

**Comment by Frenzie** [comment]
There's no need to add it to everything, just the ones that are supposed to be identical to the source. Which right now is only fr, nl and pl I think? You need to look at the checks to see what's actually failing: https://travis-ci.org/github/FreshRSS/FreshRSS/jobs/752642159

I think it'd be better to revert the ignore commits for the other languages. I can probably do that if desired, but it'd have to wait a while.

**Comment by rom-1** [comment]
ok, reverted all commits except fr, nl and pl, as suggested by Frenzie

File: cli/i18n/ignore/pl.php [hunk 6]
```diff
@@ -1,6 +1,8 @@
 <?php
 
 return array(
+	'gen.date.format_date',
+	'gen.date.format_date_hour',
```
Comment (Frenzie) [inline cli/i18n/ignore/pl.php]: We'll have to wait a bit for Travis, but I suspect/hope it'll complain this isn't alphabetical. ;-)

**Review by Alkarex — APPROVED** [review]

**Pull request merged**